language/stdlib/modules/Diem.move

```diff
@@ -500,6 +518,35 @@ module Diem {
 
     }
 
+
+
+    ///////////////////////////////////////////////////////////////////////////
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Upgrade to Concurrent Preburns for DDs
+    ///////////////////////////////////////////////////////////////////////////
+
+    // when dd tries to preburn
+    public fun remove_preburn_singleton_resource_if_exists<CoinType>(account: &signer) acquires Preburn<CoinType> {
+        let sender = Signer::address_of(account);
+        if (exists<Preburn<CoinType>>(sender)) {
+            let Preburn<CoinType> {} = move_from<Preburn<CoinType>>(sender);
```

Comment:
return preburn that you're moving out of storage (then pass into concurrent
if nonzero value then it'll fail (which might be fine since current api would fail anyway)

Reply:
destroy zero on inner coin needed Diem::